xml/signals/basic-enhanced/appearance-one-searchlight-high.xml

```diff
@@ -0,0 +1,178 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet href="../../XSLT/appearancetable.xsl" type="text/xsl"?>
+
+<appearancetable
+  xmlns:xsi = "http://www.w3.org/2001/XMLSchema-instance"
+  xsi:noNamespaceSchemaLocation = "http://jmri.org/xml/schema/appearancetable.xsd">
+
+  <copyright xmlns="http://docbook.org/ns/docbook">
+    <year>2009</year>
+    <year>2010</year>
+    <year>2024</year>
+      <holder>JMRI</holder>
+  </copyright>
+  
+  <authorgroup xmlns="http://docbook.org/ns/docbook">
+    <author>
+      <personname>
+        <firstname>John</firstname>
+        <surname>Purcell</surname>
+      </personname>
+    </author>
+
+    <author>
+      <personname>
+        <firstname>Dick</firstname>
+        <surname>Bronson</surname>
+      </personname>
+    </author> 
+
+    <author>
+      <personname>
+        <firstname>Bob</firstname>
+        <surname>Jacobsen</surname>
+      </personname>
+      <email>jake@physics.berkeley.edu</email>
+    </author>
+
+  </authorgroup>
+    
+  <revhistory xmlns="http://docbook.org/ns/docbook">
+
+    <revision>
+      <revnumber>1</revnumber>
+      <date>2009-12-28</date>
+      <authorinitials>RGJ</authorinitials>
+      <revremark>Initial version</revremark>
+    </revision>
+
+    <revision>
+      <revnumber>1.1</revnumber>
+      <date>2011-06-13</date>
+      <authorinitials>KWD</authorinitials>
+      <revremark>Add Aspect Mapping and Specific Aspects</revremark>
+    </revision>
+
+    <revision>
+      <revnumber>1.2</revnumber>
+      <date>2011-06-13</date>
+      <authorinitials>JDP</authorinitials>
+      <revremark>Revise for use in new "basic-enhanced" signal system</revremark>
+    </revision>
+
+  </revhistory>
+
+  <aspecttable>basic-enhanced</aspecttable>
+  <name>Single searchlight high signal</name>
+  <reference>(similar to basic signals described in various magazine articles)</reference>
+  <description>A simple signal for use as e.g. a block signal</description>
+  
+  <appearances>
+
+    <appearance>
+      <aspectname>Clear</aspectname>
+      <show>green</show>
+      <imagelink>../../../resources/icons/smallschematics/aspects/basic-enhanced/SL-1-high/SL-1-high-G.gif</imagelink>
+    </appearance>
+
+    <appearance>
+      <aspectname>Approach</aspectname>
+      <show>yellow</show>
+      <imagelink>../../../resources/icons/smallschematics/aspects/basic-enhanced/SL-1-high/SL-1-high-Y.gif</imagelink>
+    </appearance>
+
+    <appearance>
+      <aspectname>Advanced Approach</aspectname>
+      <show>flashyellow</show>
+      <imagelink type="aspects">../../../resources/icons/smallschematics/aspects/basic-enhanced/SL-1-high/SL-1-high-FY-animated.gif</imagelink>
+      <imagelink type="noflash">../../../resources/icons/smallschematics/aspects/basic-enhanced/SL-1-high/SL-1-high-FY-static.gif</imagelink>
```

Comment:
@bobjacobsen @dsand47 
Headless CI fails on:
`LoadAndStoreAllSignalSystemsTest.tearDown:391 Unexpected ERROR or higher messages emitted: ERROR: image type noflash does not exists, xml/signals/basic-enhanced/appearance-one-searchlight-high.xml`

I think it's this line it complains about. But I don't know enough about signal xml files to know how to fix it.

But as an example, the file `xml/signals/sample-aspects.xml` has an `imagetypes` section:
```
  <imagetypes>
    <imagetype type="large" />
    <imagetype type="small" />
  </imagetypes>
```

Reply:
This is coming from line 88 in `appearance-one-searchlight-high.xml` file

```      <imagelink type="noflash">../../../resources/icons/smallschematics/aspects/basic-enhanced/SL-1-high/SL-1-high-FY-static.gif</imagelink>
```

That's the only `imagelink` in this signal system which has the `noflash` attribute value.

The `aspects.xml` file is missing an `imagetypes` element, something like this:

```
  <imagetypes>
    <imagetype type="aspects" />
    <imagetype type="noflash" />
  </imagetypes>
```

That's the section from the BNSF-1996 signal system definition, which uses `noflash` values extensively. So the fix is to add that a section like that above between the end of the `aspects` element and the beginning of the `appearancefiles` section of basic-enhanced/aspects.xml, at line 145:

```
  </aspects>

  <imagetypes>
    <imagetype type="aspects" />
    <imagetype type="noflash" />
  </imagetypes>

  <appearancefiles>
```

Reply:
Hi Bob:  Thank you for your help.  This brings up an interesting question.  Should all the <imagelink> tags have the a"type=.." attribute?  I have made the correction you suggested to the aspects.xml file.  I can also update all the mast files to update the <imagelink> tags.  To maintain consistency I would like to either use the type attribute in all mast file, or none.  Please advise.  --John.

Reply:
I don't know what that attribute does!  I've scanned the existing files for it.  Most signal systems don't use it.  A few do:  UP-2008, SPTCO-1969 and some European ones.  Maybe somebody else can comment on the attribute.

Reply:
When you want to have alternate icons, you need to use the `<imagelink type="noflash">` attribute.  This needs to be used for at least the `<appearance>` that has an alternate icon.

You also need the image types entry in the aspects.xml file.
```
  <imagetypes>
    <imagetype type="noflash" />
  </imagetypes>
```

When you add the signal mast icon to the panel, you choose which icon set to use.

![Screenshot 2024-11-12 at 6 49 50 PM](https://github.com/user-attachments/assets/84dcb6f3-5355-4701-9130-ecfea7b9c0c0)

There is always a **Default** selection which I think is the first entry in the appearance file for each aspect.

Since you like consistency, I recommend adding the **aspect** image type to the image types list and used it for the default types for each image link.

This not actually change anything since **Default** and **aspect** refer to the same image links.

Reply:
Just uploaded revised files.  All appearance files have <imagelink> tags all modified with type= attribute, <imagetypes> section added to aspects.xml file.  Cleaned up revision history in appearance files for correct date and to show this as initial release, while preserving previous author contributions.  Edited index.shtml file to eliminate tab characters, and changed prose for consistency in terms and (hopefully) clarity.  Note these updates resulted in changes to all files in xml/signals/basic-enhanced directory.